OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -605,8 +605,10 @@ public void ResolveOrder(Actor self, Order order)
 				{
 					if (IsPlane)
 						self.QueueActivity(new ReturnToBase(self, Info.AbortOnResupply));
-					else
+					else if (!Info.RepairBuildings.Contains(order.TargetActor.Info.Name))
```

Comment:
This isnt really the right place to do this.  The check should be inside both `HeliReturnToBase` *and* `ReturnToBase` because this is an implementation detail of *where* in the base to return to, and the issue applies to both helicopters and planes.